File: python/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        
        div = 1
        while x >= 10 * div:
            div *= 10
            
        while x:
            right = x % 10
            left = x // div
            
            if left != right: return False
            
            x = (x % div) // 10
            div = div / 100
        return True

File: c/0009-palindrome-number.c
bool isPalindrome(int x){
    if (x < 0) {
        return false;
    }
    long div = 1;
    while (x >= 10 * div) {
        div *= 10;
    }

    while (x) {
        int right = x % 10;
        int left = x / div;

        if (left != right) {
            return false;
        }

        x = (x % div) / 10;
        div /= 100;
    }

    return true;
}